pre_commit_hooks/ros_include_guard.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2022 Tier IV, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import pathlib
+
+
+class DirectiveBase:
+    def __init__(self, directive):
+        self.directive = directive
+        self.line = None
+        self.text = None
+        self.expected = None
+        self.tokens = None
+
+    def update(self, line, text):
+        self.line = line
+        self.text = text
+
+    def is_none(self):
+        return self.line is None
+
+    def mismatch(self):
+        return self.text != self.expected
+
+    def overwrite(self, lines):
+        lines[self.line] = self.expected
+
+
+class OpeningDirective(DirectiveBase):
+    def prepare(self, macro_name):
+        self.tokens = split_space_boundary(self.text)
+        if 3 <= len(self.tokens):
+            self.tokens[2] = macro_name
+            self.expected = "".join(self.tokens)
+        else:
+            self.expected = f"{self.directive} {macro_name}"
+
+
+class ClosingDirective(DirectiveBase):
+    def prepare(self, macro_name):
+        self.tokens = split_space_boundary(self.text)
+        if 5 <= len(self.tokens):
+            self.tokens[4] = macro_name
+            self.expected = "".join(self.tokens)
+        else:
+            self.expected = f"{self.directive}  // {macro_name}"
+
+
+class IncludeGuard:
+    def __init__(self):
+        self.ifndef = OpeningDirective("#ifndef")
+        self.define = OpeningDirective("#define")
+        self.endif = ClosingDirective("#endif")
+        self.pragma = False
```

Comment:
To show it's `bool` and for `#pragma once`, how about this?

```suggestion
        self.has_pragma_one = False
```

Reply:
https://github.com/tier4/pre-commit-hooks-ros/pull/32/commits/f32eaf547b6dc435af99df2a9b7fcc36149cc934